openedx/core/djangoapps/user_api/accounts/api.py

```diff
@@ -643,10 +643,12 @@ def _validate_password(password, username=None, email=None, reset_password_page=
 
     # TODO: VAN-666 - Restrict this feature to reset password page for now until it is
     #  enabled on account sign in and register.
-    if settings.ENABLE_AUTHN_RESET_PASSWORD_HIBP_POLICY and reset_password_page:
+    if (settings.ENABLE_AUTHN_RESET_PASSWORD_HIBP_POLICY and reset_password_page) or \
+            (settings.ENABLE_AUTHN_REGISTER_HIBP_POLICY and not reset_password_page):
         pwned_response = check_pwned_password(password)
         if pwned_response.get('vulnerability', 'no') == 'yes':
-            raise errors.AccountPasswordInvalid(accounts.AUTHN_PASSWORD_COMPROMISED_MSG)
+            if reset_password_page or pwned_response.get('frequency', 0) >= settings.HIBP_PASSWORD_FREQUENCY_THRESHOLD:
+                raise errors.AccountPasswordInvalid(accounts.AUTHN_PASSWORD_COMPROMISED_MSG)
```

Comment:
The initial condition `settings.ENABLE_AUTHN_REGISTER_HIBP_POLICY and not reset_password_page` will evaluate `True` even for sign in page which is not what we want. 

Also should we separate the reset password and register page condition to make it more legible? This will help us in the future if we decide to change the error message for registration or reset password step.

 And maybe move it to a separate function that we can call in `_validate_password`.
```python
def _check_password_vulnerability():
    pwned_response = check_pwned_password(password)
    if pwned_response.get('vulnerability', 'no') == 'no':
        return

    if settings.ENABLE_AUTHN_RESET_PASSWORD_HIBP_POLICY and reset_password_page:
        raise errors.AccountPasswordInvalid(accounts.AUTHN_PASSWORD_COMPROMISED_MSG)
    elif (
        settings.ENABLE_AUTHN_REGISTER_HIBP_POLICY and
        registration_page and
        pwned_response.get('frequency', 0) >= settings.HIBP_PASSWORD_FREQUENCY_THRESHOLD
    ):
        raise errors.AccountPasswordInvalid(accounts.AUTHN_PASSWORD_COMPROMISED_MSG)
```

Reply:
> The initial condition settings.ENABLE_AUTHN_REGISTER_HIBP_POLICY and not reset_password_page will evaluate True even for sign in page which is not what we want.

We are using this validation function `_validate_password` only for Authn MFE's register and reset password page's password field validation. For Sign In, the API endpoint and its validation is separate

> Also should we separate the reset password and register page condition to make it more legible? This will help us in the future if we decide to change the error message for registration or reset the password step.

As discussed, the error message for all the scenarios (HIBP) is going to be the same